pizzalab/pizzalabapp/views.py

```diff
@@ -1,42 +1,76 @@
 from django.contrib.auth import get_user_model
-from rest_framework import viewsets
+from rest_framework import status, viewsets
 from rest_framework.authentication import SessionAuthentication
 from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
 
 from pizzalabapp.models import Ingredient, Order, OrderItem, Pizza
 from pizzalabapp.serializers import (IngredientSerializer, OrderItemSerializer,
                                      OrderSerializer, PizzaSerializer)
 
 
 class IngredientViewSet(viewsets.ModelViewSet):
+    """
+    Viewset for Ingredient objects
+    """
     queryset = Ingredient.objects.all()
     serializer_class = IngredientSerializer
 
 
 class PizzaViewSet(viewsets.ModelViewSet):
+    """
+    Viewset for Pizza objects
+    """
     queryset = Pizza.objects.all()
     serializer_class = PizzaSerializer
 
 
 class OrderViewSet(viewsets.ModelViewSet):
-    queryset = Order.objects.all()
+    """
+    Viewset for Order objects
+    create and get_queryset methods are overridden
+    """
     serializer_class = OrderSerializer
     authentication_classes = [SessionAuthentication]
     permission_classes = [IsAuthenticated]
 
-    def perform_create(self, serializer):
+    def create(self, request, *args, **kwargs):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
         customer_requested = self.request.data.get('customer')
-        if self.request.user.is_staff and customer_requested:
+        user = self.request.user
+
+        if customer_requested and not user.is_staff:
+            return Response(
+                "Ordering for other customers is prohibited",
+                status.HTTP_400_BAD_REQUEST
+            )
+
+        customer = customer_requested if customer_requested and user.is_staff \
+            else user
```

Comment:
Didn't realize that customer_requested and user aren't same type and it doesn't really make sense, so even if we load this a bit more, I think we should have something like 
customer = get_user_model().objects.get(id=customer_requested)\
            if customer_requested and user.is_staff else user

Which also gets us rid of the if else below: serializer.save(customer=customer) should suffice